extensions/interactions/NumericInput/directives/NumericInput.js

```diff
@@ -49,6 +51,14 @@ oppia.directive('oppiaInteractiveNumericInput', [
               });
             }
           };
+
+          $scope.isWindowNarrow = function() {
+            return WindowDimensionsService.isWindowNarrow();
+          };
+
+          $scope.$on('progress-nav-submit', function() {
```

Comment:
Please file an issue to find a way to do this transparently -- i.e. we should have some common init() that is called for all interactions, and that init() should register the event handler if the interaction dict has an includes_submit_button attribute (see above).